roles/matrix-dynamic-dns/tasks/validate_config.yml

```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Fail if required settings not defined
+  fail:
+    msg: >-
+      You need to define a required configuration setting (`{{ item }}`).
+  when: "vars[item] == ''"
+  with_items:
+    - "matrix_dynamic_dns_username"
+    - "matrix_dynamic_dns_password"
+    - "matrix_domain"
```

Comment:
Perhaps a new variable (`matrix_dynamic_dns_domain`), with an empty default value would be better.

We can then define that in terms of `matrix_domain` in `group_vars/matrix_servers`:

```yaml
matrix_dynamic_dns_domain: "{{ matrix_domain }}"
```